musicbrainz-postgres/scripts/DBDefs.pm

```diff
@@ -0,0 +1,481 @@
+#!/usr/bin/env perl
+
+use warnings;
+# vi: set ts=4 sw=4 :
+#____________________________________________________________________________
+#
+#   MusicBrainz -- the open internet music database
+#
+#   Copyright (C) 1998 Robert Kaye
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation; either version 2 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+#   $Id$
+#____________________________________________________________________________
+
+package DBDefs;
+use parent 'DBDefs::Default';
+use MusicBrainz::Server::Replication ':replication_type';
+use MusicBrainz::Server::DatabaseConnectionFactory;
+
+# Commented-out lines should generally have sane defaults; anything that's uncommented
+# probably needs personal attention.
+
+################################################################################
+# Directories
+################################################################################
+
+# The server root, i.e. the parent directory of admin, bin, lib, root, etc.
+# By default, this uses the path of lib/DBDefs/Default.pm, minus '/lib/DBDefs/Default.pm'
+# sub MB_SERVER_ROOT    { "/home/httpd/musicbrainz/musicbrainz-server" }
+# Where static files are located
+# sub STATIC_FILES_DIR { my $self= shift; $self->MB_SERVER_ROOT . '/root/static' }
+
+################################################################################
+# The Database
+################################################################################
+
+# Configuring databases here is required; there are no defaults.
+MusicBrainz::Server::DatabaseConnectionFactory->register_databases(
+    # How to connect when we need read-write access to the database
+    READWRITE => {
+        database    => "musicbrainz",
+        schema      => "musicbrainz",
+        username    => "musicbrainz",
```

Comment:
In the next iteration, should probably do sed replace on username/pass based on incoming ENV variables